### DISCUSSION TITLE
Is tycho-surefire-plugin honoring the p2.inf instructions when testRuntime=default ?

### DISCUSSION THREAD

**Discussion by henryju** [eclipse-tycho/tycho#3440] [Q&A] [answered]
Hi,

We are trying to use platform-specific fragments in our plugin. Since those fragments include executable binaries, we have this idea of avoiding the cost of extraction in a temporary folder at every startup by asking Eclipse to install the bundle "exploded" (Eclipse-BundleShape: dir).
Since the ZIP/JAR format doesn't support permissions, I have added a p2.inf file in each fragment to restore permissions at installation time:

```
instructions.install = \
    chmod(targetDir:@artifact,targetFile:bin,permissions:755,options:-R);
 instructions.install.import= \
    org.eclipse.equinox.p2.touchpoint.eclipse.chmod
```

This is working fine when doing manual testing, but not during the JUnit tests. The fragment is correctly exploded by Tycho, but I have the impression that p2.inf instructions are ignored.
I gave a try to change to `<testRuntime>p2Installed</testRuntime>` and now the permissions are correctly set in the exploded bundle (but I have other failures now).

I read that `<testRuntime>p2Installed</testRuntime>` is experimental, so my question is: am I forced to use this mode if I want the chmod instruction to be executed?

**Comment by laeubi** [comment, marked as answer]
@henryju p2.inf touchpoints are only honored when using the `<testRuntime>p2Installed</testRuntime>`

> I read that <testRuntime>p2Installed</testRuntime> is experimental

I think it is a bit misleading, but for sure it is not used very often and a rather special case like this.

> permissions:755

By the way making is readable and executable for everyone is probably a security risk

**Reply by henryju** [reply]
Thanks @laeubi 

I have changed to `permissions:+x` as it is done in [justj](https://github.com/eclipse-justj/justj.tools/blob/f634d765adc26df6ebd87135645fbd732e18bcca/plugins/org.eclipse.justj.codegen/src/org/eclipse/justj/codegen/model/util/Reconciler.java#L438C80-L438C94).

Now I need to find how to make the tests run with `p2Installed`, or maybe use a workaround to programmatically change the file permission in the tests itself.

**Reply by laeubi** [reply]
If you can share what is not working we probably can work it out, as said this mode of operation has some caveats but it is used for example in PDE build, there is one test for it so it _should_ work (using latest 4.0.5 Tycho release) without any special consideration.

**Reply by henryju** [reply]
The error I get is:
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:4.0.5:test (default-test) on project org.sonarlint.eclipse.core.tests: Error while executing platform: The launcher bundle org.eclipse.equinox.launcher_*.jar was not found in /home/julien.henry/Projects/sonarlint/sonarlint-eclipse/org.sonarlint.eclipse.core.tests/target/work/plugins
```

**Reply by laeubi** [reply]
Can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue? Are you using latest Tycho release? Anything special (e.g. running on mac or similar?)....

**Reply by henryju** [reply]
I am using Tycho 4.0.5, on Linux. Since my project is a feature and not a full product, I have the impression I need somehow to tell Tycho what product to "materialize" to create the test environment.
I tried to add:
`<profileName>epp.package.jee</profileName>`
but it makes no difference.
Maybe I should change something in my target platform.

I can try to write an integration test.

**Reply by laeubi** [reply]
You don't necessarily should need a product, maybe you can try to specify the launcher as an explicit dependency but I think the best would be if tycho can figure it out automatically so probably a good opportunity to improve this feature.

The profile name could be indeed a problem if you are referencing a full product.

**Reply by henryju** [reply]
> maybe you can try to specify the launcher as an explicit dependency

How would you do that?

I tried to add:
```
          <plugin>
            <groupId>org.eclipse.tycho</groupId>
            <artifactId>tycho-surefire-plugin</artifactId>
            <configuration>
              <dependencies>
		<dependency>
                  <type>p2-installable-unit</type>
                  <artifactId>org.eclipse.equinox.launcher</artifactId>
                </dependency>
```
but it seems to have no effect.
